### PR TITLE
TE5-T637: Practical-04

### PR DESCRIPTION
**Pull request by JeetMakadiya**

### CODE REVIEW

**Comment by JeetMakadiya** [comment]
Add, Edit, Delete card functionality using API.

File: practical-04/src/components/HomePage.vue [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    },
+    // Func. to get card data from API.
+    async getCarData() {
+      try {
+        let response = await Axios.get(
+          "https://testapi.io/api/dartya/resource/cardata"
+        );
+        await this.formatCarData(response.data.data);
+      } catch (error) {
+        console.log(error);
```
Comment (rajbhojani88) [inline practical-04/src/components/HomePage.vue]: this.errmsg = "oops! somthing went wrong";

File: practical-04/src/components/HomePage.vue [hunk 2]
```diff
@@ -0,0 +1,199 @@
+<template>
+  <div>
+    <NavBar />
+    <section>
+      <b-container class="bv-example-row">
```
Comment (rajbhojani88) [inline practical-04/src/components/HomePage.vue]: class ?

File: practical-04/src/components/HomePage.vue [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+      } else {
+        // else for Add New Card
+        await this.addCarData(data);
+        await this.getCarData();
+      }
+    },
+    // Func. to get card data from API.
+    async getCarData() {
+      try {
+        let response = await Axios.get(
```
Comment (rajbhojani88) [inline practical-04/src/components/HomePage.vue]: check for success

File: practical-04/src/components/HomePage.vue [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+          "https://testapi.io/api/dartya/resource/cardata"
+        );
+        await this.formatCarData(response.data.data);
```
Comment (rajbhojani88) [inline practical-04/src/components/HomePage.vue]: check is res has data

File: practical-04/src/components/HomePage.vue [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+          description: item.details,
+          price: item.price,
+        };
+      });
+    },
+    // Func. to ADD card data to API.
+    async addCarData(data) {
+      this.isLoading = true;
+      try {
+        await Axios.post("https://testapi.io/api/dartya/resource/cardata", {
```
Comment (rajbhojani88) [inline practical-04/src/components/HomePage.vue]: use axios
        .post({
        })
        .then((res) => {
        })
        .catch((err) => {
        });

**Pull request merged**